src/renderer/src/hooks/useGetFullTransactions.tsx

```diff
@@ -0,0 +1,316 @@
+          ...item,
+          blockchain,
+          isPending: false,
+          events: events.map(({ from, to, ...event }) => ({
+            ...event,
+            from,
+            to,
+            fromAccount: from ? allAccounts.find(AccountHelper.predicate({ address: from, blockchain })) : undefined,
+            toAccount: to ? allAccounts.find(AccountHelper.predicate({ address: to, blockchain })) : undefined,
+          })),
+        })
+      })
+    } catch {
+      /* empty */
+    }
+
+    const newQuery = queryCache.build(queryClient, defaultedQueryOptions) as Query<
+      TFullTransactionsByAddressResponse,
+      Error,
+      TFullTransactionsByAddressResponse,
+      string[]
+    >
+
+    newQuery.setData(queryData, { manual: true })
+
+    data.push(...queryData.data)
+
+    hasNextPage ||= !!queryData.nextCursor
+  })
+
+  await Promise.allSettled(promises)
+
+  return { data, page: hasNextPage ? page + 1 : undefined }
+}
+
+export const useGetFullTransactions = ({ accounts, dateFrom, dateTo }: TProps) => {
+  const queryClient = useQueryClient()
+  const { accountsRef } = useAccountsSelector()
+  const { networkByBlockchain: networksByBlockchain } = useSelectedNetworkByBlockchainSelector()
+  const { pendingTransactions } = usePendingTransactionsSelector()
+  const { hiddenTokensByBlockchain } = useHiddenTokensByBlockchainSelector()
+
+  const allAccounts = accountsRef.current
+
+  const query = useInfiniteQuery({
+    queryKey: buildGetFullTransactionsAggregatedQueryKey({
+      accounts,
+      dateFrom,
+      dateTo,
+      networksByBlockchain,
+    }),
+    queryFn: ({ pageParam: page }) =>
+      getFullTransactions({
+        queryClient,
+        dateFrom,
+        dateTo,
+        accounts,
+        allAccounts,
+        networksByBlockchain,
+        page,
+      }),
+    initialPageParam: 1,
+    getNextPageParam: ({ page }) => page,
+  })
+
+  const data = useMemo(() => {
+    const filteredPendingTransfersByAccounts = pendingTransactions.filter(
+      ({ account, time }) =>
+        accounts.some(AccountHelper.predicate(account)) &&
+        isWithinInterval(createNewDateByTime(time), { start: dateFrom, end: dateTo })
+    )
+
+    const groupedPendingTransfersByHash = groupBy(filteredPendingTransfersByAccounts, 'hash')
+    const items = query.data?.pages?.flatMap(page => page?.data ?? []) ?? []
+
+    const filteredItems = items.filter(
+      ({ txId }, index, array) =>
+        !groupedPendingTransfersByHash[txId] && array.findIndex(item => item.txId === txId) === index
+    )
+
+    filteredItems.unshift(
+      ...Object.values(groupedPendingTransfersByHash).map(transfers => {
+        transfers.sort(({ time: newerTime }, { time: olderTime }) => newerTime - olderTime)
+
+        const [firstTransfer] = transfers
+
+        const {
+          hash,
+          account: { blockchain },
+        } = firstTransfer
+
+        const transaction = items.find(({ txId }) => txId === hash)
```

Comment:
Just complementing the comment above. 

I think I don't need to take the transaction into consideration, it is justing letting the code more complex and it barely changes the user experience